src/server/mockServer.js

```diff
@@ -0,0 +1,54 @@
+const path = require('path');
+const express = require('express');
+const mockServer = express();
+
+const port = 9990;
+
+// TODO: move these functions to a middleware file?
+// TODO: create a function and endpoint to delete a mock route
+
+// this function gets called when the /mock endpoint is hit and is what sends the mock response
+const createMockRoute = (method, endpoint, response) => {
+  mockServer[method.toLowerCase()](endpoint, (req, res) => {
+    res.send(response);
+  });
+};
+
+// returns all existing routes as an array of strings
+const getAllExistingRoutes = () => {
+  const routes = mockServer._router.stack
+    .filter((layer) => layer.route)
+    .map(
+      (layer) => `${Object.keys(layer.route.methods)} - ${layer.route.path}`
+    );
+
+  return routes;
```

Comment:
2 things:

1. SUPER nit:

```suggestion
  return mockServer._router.stack
    .filter((layer) => layer.route)
    .map(
      (layer) => `${Object.keys(layer.route.methods)} - ${layer.route.path}`
    );
```

2. What is the difference between using `filter` vs. just doing `mockServer._router.stack.route`?

Reply:
1. In my opinion, the separated return statement is easier to read and maintain. "routes" or perhaps a more descriptive "arrayOfRoutes" is easier to understand vs "return mockServer._router.stack.filter().map()" because it is not immediately apparent what that is. I feel like this is fine as is?
2. `mockServer._router.stack` is an array of objects, not an object (which is what I think you are getting at with the .route thing...?) where I could use the dot operator to access the routes. The function filters for objects that have a route property and maps a string so we can see the method and the path. Let me know if there's something specific I can clear up for you, unless this resolved your question.